jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js

```diff
@@ -0,0 +1,93 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const PropTypes = require('prop-types');
+const React = require('react');
+const classNames = require('classnames');
+
+const GenericButton = require('../../generic-button/generic-button');
+const TermsPopup = require('../../terms-popup/terms-popup');
+
+class DeploymentAgreements extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showTerms: false
+    };
```

Comment:
We could bind _toggleTerms here and avoid doing it every time below? [shrug]

Reply:
I think we looked at that pattern when we switched over to classes, but it makes it slightly harder to debug as you have to remember to check that functions have been bound and can result in some strange errors. Happy to revisit if people are keen.